src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java

```diff
@@ -1003,6 +1003,31 @@ public void onJailedPlayerUsesCommand(PlayerCommandPreprocessEvent event) {
 		}
 	}
 	
+	/** 
+	 * Allows restricting commands while being on an town.
+	 * Works almost the same way as jail command blacklisting.
+	 * @param event
+	 */
+	@EventHandler(priority = EventPriority.NORMAL)
+	public void onPlayerUsesCommandInsideTown(PlayerCommandPreprocessEvent event) {
+		if (plugin.isError())
+			return;
+		
```

Comment:
Requires 
```java
		if (!TownyAPI.getInstance().isTownyWorld(event.getPlayer().getWorld()))
			return;
		
		if (TownyAPI.getInstance().isWilderness(event.getPlayer().getLocation()))
			return;
```
after the plugin.isError() check.

One thing that should be noted is that the main thing this will be used for is the /sethome command, and this will end up blocking the use of that command even in personally owned plots where its probably safe to have it. Sethome is a bit of a special case because this PR wouldn't stop someone from using /sethome directly outside of someone's Town either. 

It might be a good idea to have another listener specifically targeting things like /sethome and limiting it's use to personally-owned plots, with a configurable list so admins can limit specific commands to only plots that a resident owns personally.

Reply:
github keeps bugging me about this, but this was already implemented